apps/google-analytics-4/frontend/src/components/main-app/analytics-app/AnalyticsApp.tsx

```diff
@@ -41,30 +57,20 @@ const AnalyticsApp = () => {
     }
 
     fetchData();
-  }, []);
+  }, [api, reportRequestParams]);
 
   useEffect(() => {
-    const sliceByDateRange = (dateRange: DateRangeType): RunReportResponse => {
-      let newRows: Row[] = [];
-      switch (dateRange) {
-        case DateRange.LastDay:
-          newRows = runReportResponse.rows.slice(0, 2);
-          break;
-        case DateRange.LastWeek:
-          newRows = runReportResponse.rows.slice(0, 7);
-          break;
-        case DateRange.LastMonth:
-          newRows = runReportResponse.rows.slice(0, 28);
-          break;
-      }
-      return { ...runReportResponse, rows: [...newRows] };
-    };
     if (runReportResponse.rowCount) {
-      const newData = sliceByDateRange(dateRange);
-      setPageViewData(newData);
+      setStartEndDates(getRangeDates(dateRange));
+      setRunReportResponse(runReportResponse);
+      setLoading(false);
     }
   }, [dateRange, runReportResponse]);
 
+  useEffect(() => {
```

Comment:
I think we can get rid of this useEffect and just do the logic in the `handleDateRangeChange` function because the only time the dateRange changes is when the user does an action right?

Reply:
absolutely, love the idea of consolidating this 💯